fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/LuceneSearchProvider.java

```diff
@@ -317,6 +323,114 @@ private void updateDimensionRow(
         writer.updateDocument(keyTerm, luceneDimensionRowDoc);
     }
 
+    @Override
+    public void replaceIndex(String newLuceneIndexPathString) {
+        lock.writeLock().lock();
+        try {
+            Path oldLuceneIndexPath = Paths.get(luceneIndexPath);
+            String tempDir = oldLuceneIndexPath.resolveSibling(oldLuceneIndexPath.getFileName() + "_old").toString();
+
+            LOG.debug("Moving old Lucene index directory from {} to {} ...", luceneIndexPath, tempDir);
+            moveDirEntries(luceneIndexPath, tempDir);
+
+            LOG.debug("Moving all new Lucene indexes from {} to {} ...", newLuceneIndexPathString, luceneIndexPath);
+            moveDirEntries(newLuceneIndexPathString, luceneIndexPath);
+
+            LOG.debug(
+                    "Deleting {} since new Lucene indexes have been moved away from there and is now empty",
+                    newLuceneIndexPathString
+            );
+            deleteDir(newLuceneIndexPathString);
```

Comment:
?

Reply:
Once we move all new indexes, the directory used to contain them is empty. So I delete it